application/controllers/Hook.php

```diff
@@ -0,0 +1,254 @@
+        //create URL
+        $url = sprintf("https://storage.googleapis.com/prestashop-core-nightly/reports/%s", $filename);
+        log_message('info', "creating url $url");
+        //retrieve content
+        log_message('info', "retrieving content...");
+        try {
+            $contents = file_get_contents($url);
+        } catch(Exception $e) {
+            log_message('error', "Could not retrieve content from $url");
+            header($_SERVER['SERVER_PROTOCOL'] . ' 400 Bad Request', true, 400);
+            exit("unable to decode JSON data");
+        }
+
+        log_message('info', "decoding JSON...");
+        try {
+            $file_contents = json_decode($contents);
+        } catch(Exception $e) {
+            log_message('error', "Could not decode JSON data from the file");
+            header($_SERVER['SERVER_PROTOCOL'] . ' 400 Bad Request', true, 400);
+            exit("unable to decode JSON data");
+        }
+
+        //retrieving version number
+        preg_match($this->pattern, $filename, $matches);
+        if (!isset($matches[1])) {
+            exit("could not retrieve version from filename '$filename'");
+        }
+        $version = $matches[1];
+        if (strlen($matches[1]) < 1) {
+            header($_SERVER['SERVER_PROTOCOL'] . ' 400 Bad Request', true, 400);
+            exit("version found not correct ('$version') from filename '$filename'");
+        }
+
+        //starting real stuff
+        //create the execution
+        $stats = $file_contents->stats;
+        $execution_data = [
+            'ref' => date('YmdHis'),
+            'start_date' => date('Y-m-d H:i:s', strtotime($stats->start)),
+            'end_date' => date('Y-m-d H:i:s', strtotime($stats->end)),
+            'duration' => $stats->duration,
+            'version' => $version
+        ];
+        //let's check if there's not a similar entry...
+        $entry_date = date('Y-m-d', strtotime($stats->start));
+        $similar = $this->Execution->findSimilarEntries($entry_date, $version);
+
+        if ($similar !== NULL) {
+            if (!$this->force) {
+                log_message('error', "A similar entry was found (criteria: version $version and date $entry_date)");
+                header($_SERVER['SERVER_PROTOCOL'] . ' 409 Conflict', true, 409);
+                exit("A similar entry was found (criteria: version '$version' and date '$entry_date'). Use the 'force' parameter to force insert");
+            } else {
+                log_message('warning', "A similar entry was found (criteria: version $version and date $entry_date) but FORCING insert anyway");
+            }
+        }
+
+        try {
+            log_message('info', "Inserting Execution");
+            $this->execution_id = $this->Execution->insert($execution_data);
+        } catch(Exception $e) {
+            header($_SERVER['SERVER_PROTOCOL'] . ' 500 Internal Server Error', true, 500);
+            exit("could not insert execution");
```

Comment:
```suggestion
            exit("failed insert execution");
```
`could not` = `there was a check or a if and the execution was stopped`
`failed` = `I tried but got an error`

Reply:
Semantic details are tight